Assets/BossRoom/Scripts/Client/Game/Character/ClientInputSender.cs

```diff
@@ -169,12 +176,30 @@ void FixedUpdate()
                 {
                     m_LastSentMove = Time.time;
                     var ray = m_MainCamera.ScreenPointToRay(Input.mousePosition);
-                    if (Physics.RaycastNonAlloc(ray, k_CachedHit, k_MouseInputRaycastDistance, k_GroundLayerMask) > 0)
+
+                    // clear previous results
+                    for (int i = 0; i < k_CachedHit.Length; i++)
```

Comment:
This for loop shouldn't be necessary if you just iterate over the first items until you reach "groundHits". with non-alloc physics calls, you really don't need to clear anything from one run to another. As long as you iterate from 0 to the returned number of items, you should be fine

Reply:
This is correct. Old `RaycastHit`s don't get cleared, but the sorting I did below accounted for this anyways.

Reply:
oh wait true, there's a sort. How does it take it into account? It just orders by distance. So if an old collision is closer, wouldn't get still be included in this?

Reply:
You can pass an `index` and a `length` to the sort method. So long as I only sort until `groundHits`, any hits from 0 to `groundHits` inside `k_CachedHit` will be valid, regardless of what was searched the previous occurrence.

Reply:
And that's what happens when you read too quickly.
thanks

Reply:
No prob!